fe/fe-core/src/main/java/com/starrocks/alter/Alter.java

```diff
@@ -374,6 +377,17 @@ public void processAlterTable(AlterTableStmt stmt) throws UserException {
         // some operations will take long time to process, need to be done outside the databse lock
         boolean needProcessOutsideDatabaseLock = false;
         String tableName = dbTableName.getTbl();
+
+        // this logic is use to adapt mysql syntax, and cannot be executed after adding a write lock
+        // ALTER TABLE test TRUNCATE PARTITION p1;
+        if (currentAlterOps.hasTruncatePartitionOp()) {
+            TruncatePartitionClause clause = (TruncatePartitionClause) alterClauses.get(0);
+            TableRef tableRef = new TableRef(stmt.getTbl(), null, clause.getPartitionNames());
```

Comment:
If somebody is creating a partition when you truncate the partition? How to handle the conflict

Reply:
This PR is just a mysql syntax compatible PR, and there is a corresponding locking and unlocking in the original truncate partition logic implemented to solve this problem